src/main/java/dev/cdevents/CDEventTypes.java

```diff
@@ -14,217 +18,922 @@ private CDEventTypes() {
-            final String buildEventType, final String buildId,
-            final String buildName, final String buildArtifactId,
-            final String buildData) {
+
+    public static CloudEvent createTaskRunStartedEvent(
+            final String eventType, final String id,
+            final URI source, final String taskName,
+            final PipelineRun pipelineRun, final URI url,
+            final String taskRunData) {
         CloudEvent ceToSend =
-                buildCloudEvent(buildEventType, buildData)
-                .withExtension("buildid", buildId)
-                .withExtension("buildname", buildName)
-                .withExtension("buildartifactid", buildArtifactId)
-                .build();
+                buildCloudEvent(eventType, taskRunData)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("taskname", taskName)
+                        .withExtension("pipelinerun", pipelineRun.toString())
+                        .withExtension("url", url)
+                        .build();
         return ceToSend;
     }
 
     /**
-     *  Creates a continuous delivery test events
+     * Creates a continuous delivery Task run finished events
      * using {@link CloudEventBuilder}
      * and returns {@link CloudEvent} object.
      *
-     * @param testEventType
-     * @param testId
-     * @param testName
-     * @param testVersion
-     * @param testData
+     * @param eventType
+     * @param id
+     * @param source
+     * @param taskName
+     * @param pipelineRun
+     * @param url
+     * @param outcome
+     * @param errors
+     * @param taskRunData
      * @return the cloudEvent object with continuous delivery
-     * test event extensions
+     * task run finished event extensions
      */
-    public static CloudEvent createTestEvent(
-            final String testEventType, final String testId,
-            final String testName, final String testVersion,
-            final String testData) {
+
+
+    public static CloudEvent createTaskRunFinishedEvent(
+            final String eventType, final String id,
+            final URI source, final String taskName,
+            final PipelineRun pipelineRun, final URI url,
+            final CDEventConstants.Outcome outcome, final String errors,
+            final String taskRunData) {
         CloudEvent ceToSend =
-                buildCloudEvent(testEventType, testData)
-                .withExtension("testid", testId)
-                .withExtension("testname", testName)
-                .withExtension("testversion", testVersion)
-                .build();
+                buildCloudEvent(eventType, taskRunData)
+                        .withExtension("id", id)
+                        .withExtension("source", source)
+                        .withExtension("taskname", taskName)
+                        .withExtension("pipelinerun", pipelineRun.toString())

```

Comment:
please use jackson-databind to write this pipelineRun as String, say objectMapper.writeValueAsString(pipelineRun)